serverless/cmd/integrate/main.go

```diff
@@ -18,48 +18,125 @@ package main
 
 import (
 	"flag"
+	"io/ioutil"
+	"os"
 
 	"github.com/golang/glog"
+	"github.com/google/trillian-examples/serverless/api"
 	"github.com/google/trillian-examples/serverless/internal/log"
 	"github.com/google/trillian-examples/serverless/internal/storage/fs"
 	"github.com/google/trillian/merkle/rfc6962/hasher"
+	"golang.org/x/mod/sumdb/note"
 
 	fmtlog "github.com/google/trillian-examples/formats/log"
 )
 
 var (
-	storageDir = flag.String("storage_dir", "", "Root directory to store log data.")
+	storageDir  = flag.String("storage_dir", "", "Root directory to store log data.")
+	initialise  = flag.Bool("initialise", false, "Set when creating a new log to initialise the structure.")
+	pubKeyFile  = flag.String("public_key", "", "Location of public key file. If unset, uses the contents of the SERVERLESS_LOG_PUBLIC_KEY environment variable.")
+	privKeyFile = flag.String("private_key", "", "Location of private key file. If unset, uses the contents of the SERVERLESS_LOG_PRIVATE_KEY environment variable.")
 )
 
 func main() {
 	flag.Parse()
 	h := hasher.DefaultHasher
+	// Read log public key from file or environment variable
+	var pubKey string
+	if len(*pubKeyFile) > 0 {
+		k, err := ioutil.ReadFile(*pubKeyFile)
+		if err != nil {
+			glog.Exitf("failed to read public_key file: %q", err)
+		}
+		pubKey = string(k)
+	} else {
+		pubKey = os.Getenv("SERVERLESS_LOG_PUBLIC_KEY")
+		if len(pubKey) == 0 {
+			glog.Exit("supply public key file path using --public_key or set SERVERLESS_LOG_PUBLIC_KEY environment variable")
+		}
+	}
```

Comment:
This code is duplicated below for the private key, would be good to factor it out, but can do that in a follow up PR (maybe just pop a TODO in?)

Reply:
Had a go. Factor out env check too?

Reply:
Yeah, good idea - could factor out a whole `getKey(filename, envName) (string, err)`, but let's do that one in a follow-up PR, I think this is good to merge for now.

Reply:
ack